src/transform/src/typecheck.rs

```diff
@@ -905,9 +909,9 @@ impl Typecheck {
             ArrangeBy { input, keys } => {
                 let t_in = tc.typecheck(input, ctx)?;
 
-                for cols in keys {
-                    for col in cols {
-                        let _ = tc.typecheck_scalar(col, expr, &t_in)?;
+                for key in keys {
```

Comment:
@mgree: Gábor referenced [the Slack thread](https://materializeinc.slack.com/archives/C02PPB50ZHS/p1676473154162599) explaining the naming fixes in the PR description.

Reply:
Also note that what was named `col` here can actually be any complex `MirScalarExpr`, not just a column reference.